server/providers/uploads/local/index.ts

```diff
@@ -0,0 +1,72 @@
+import { Provider } from '../../../uploads/types'
+import process from 'process'
+import path from 'path'
+import fs from 'fs'
+import crypto from 'crypto'
+import express from 'express'
+import config from '../../../../config/server'
+
+interface LocalProviderOptions {
+  uploadDirectory?: string;
+  endpoint: string;
+}
+
+interface Upload {
+  filePath: string;
+  name: string;
+}
+
+export default class LocalProvider implements Provider {
+  private uploadDirectory: string
+  private endpoint: string
+  private app: express.Application
+
+  private uploadMap: Map<string, Upload>
+
+  constructor (options: LocalProviderOptions, app: express.Application) {
+    if (options.uploadDirectory === undefined) {
+      options.uploadDirectory = path.join(process.cwd(), 'uploads')
+    }
+
+    if (!fs.existsSync(options.uploadDirectory)) {
+      fs.mkdirSync(options.uploadDirectory)
+    }
+
+    this.uploadDirectory = options.uploadDirectory
+    this.endpoint = options.endpoint
+    this.app = app
+
+    this.uploadMap = new Map()
+    this.app.get(this.endpoint, (req, res) => {
```

Comment:
(style) Maybe define this as an instance method instead of as an anonymous function?